### ISSUE TITLE
"LLVM ERROR: ran out of registers" when firing a weapon.

### ISSUE THREAD

**Issue by bioinfornatics** [Unvanquished/Unvanquished#701]
Dear,

game crashed while I press fire

```
GL_VENDOR: X.Org 
GL_RENDERER: Gallium 0.4 on AMD PITCAIRN 
GL_VERSION: 3.0 Mesa 10.3.3 
GL_SHADING_LANGUAGE_VERSION: 1.30 
GL_MAX_VERTEX_UNIFORM_COMPONENTS 16384 
Using GPU vertex skinning with max 256 bones in a single pass 
Extracting VM module cgame-x86_64.nexe from .unvanquished/pkg/unvanquished_0.38.0.pk3... 
Loading VM module cgame-x86_64.nexe... 
][7808,2147870592:17:31:06.216212] Native Client module will be loaded at base address 0x0000536f00000000
Loaded VM module in 50 msec 
INFO: Loaded font face Roboto Bold (from fonts/Roboto-Bold.ttf). 
INFO: Loaded font face Roboto Bold Italic (from fonts/Roboto-BoldItalic.ttf). 
INFO: Loaded font face Roboto Italic (from fonts/Roboto-Italic.ttf). 
INFO: Loaded font face Roboto Regular (from fonts/Roboto-Regular.ttf). 
INFO: Loaded font face FontAwesome Regular (from fonts/fontawesome-webfont.ttf). 
INFO: Loaded font face DejaVu Sans Condensed (from fonts/DejaVuSansCondensed.ttf). 
Bioinfornatics a rejoint la partie 
Bioinfornatics a rejoint les humans 
Repeater construit par Bioinfornatics 
]LLVM ERROR: ran out of registers during register allocation
AL lib: ReleaseALC: 1 device not closed
```

**Comment by Amanieu** [comment]
This looks like a bug in your graphics driver

**Comment by bioinfornatics** [comment]
After a check this appear only on spacetrack map

**Comment by Kangz** [comment]
Yeah this is definitely a problem in the graphics driver, we use LLVM only for loading the VMs which in succeeded in this log. @bioinfornatics if that's not too much hassle, can you take an apitrace so we can have a good repro for the Gallium people to fix this? Also updating to latest Mesa might fix that bug.

**Comment by Amanieu** [comment]
We don't even use LLVM for loading VMs, only for compiling them. The only user of LLVM in this case is your graphics driver which uses it to compile GPU shaders.

**Comment by bioinfornatics** [comment]
After downloading [apitrace](https://github.com/apitrace/apitrace) and build it

I use this command (I do not know if that is good):
```
$ LANG=C ./apitrace trace -o unvanquished.log  ~/Téléchargements/unvanquished_0.38.0/linux64/daemon
```

the output log is in binary and is huge 379 mo

**Comment by Viech** [comment]
Yes, apitrace files are always huge binaries. If you have problems uploading it, use a smaller screen resolution when testing, keep your recording short and compress the file afterwards.

**Comment by Amanieu** [comment]
You can start the game with "+map spacetracks" to skip the main menu

**Comment by Kangz** [comment]
also -set r_picmip 4 will help as the trace will contain less texture data

**Comment by bioinfornatics** [comment]
[link to compressed log for 7 days](http://82.228.212.97:8080/share/7_rtEWfZAG6bIDAx/unvanquished.log.xz)

**Comment by Viech** [comment]
@Kangz did you fetch the apitrace?

**Comment by Kangz** [comment]
Uh?

Yes now I did.

**Comment by Kangz** [comment]
A copy of the file is [here](http://inf553.kangz.net/unvanquished.log.xz
); might get deleted in a few months.

**Comment by Kangz** [comment]
The developers of the driver in #radeon explained that it was a bug in the compiler that caused temporary arrays to take more registers than they should which makes us eventually use more than the 124 GPRS (General Purpose RegisterS) of r600 hardware. As far as I understand, it is because their compiler produces badly optimized machine code, and then optimizes it which loses some information. They are discussing fixing this.

On our side we can fix this by using less arrays, and in the case replacing the deforms array by GLSL code generated on the fly.

**Comment by gimhael** [comment]
This should be fixed by commit ed8ca2cee90a277a5308c2bf898b1498bff1ec52.
